test/e2e/windows/node_shutdown.go

```diff
@@ -0,0 +1,371 @@
+		framework.ExpectNoError(err, "Error finding Windows node")
+		framework.Logf("Using node: %v", targetNode.Name)
+
+		nodeName := targetNode.Name
+		nodeSelector := fields.Set{
+			"spec.nodeName": nodeName,
+		}.AsSelector().String()
+
+		// Define test pods
+		pods := []*v1.Pod{
+			getGracePeriodOverrideTestPod("period-20-"+string(uuid.NewUUID()), nodeName, 20, ""),
+			getGracePeriodOverrideTestPod("period-25-"+string(uuid.NewUUID()), nodeName, 25, ""),
+			getGracePeriodOverrideTestPod("period-critical-5-"+string(uuid.NewUUID()), nodeName, 5, scheduling.SystemNodeCritical),
+			getGracePeriodOverrideTestPod("period-critical-10-"+string(uuid.NewUUID()), nodeName, 10, scheduling.SystemNodeCritical),
+		}
+
+		ginkgo.By("Creating batch pods")
+		e2epod.NewPodClient(f).CreateBatch(ctx, pods)
+
+		list, err := e2epod.NewPodClient(f).List(ctx, metav1.ListOptions{
+			FieldSelector: nodeSelector,
+		})
+		framework.ExpectNoError(err)
+		gomega.Expect(list.Items).To(gomega.HaveLen(len(pods)), "the number of pods is not as expected")
+
+		ctx, cancel := context.WithCancel(context.Background())
+		defer cancel()
+
+		go func() {
+			defer ginkgo.GinkgoRecover()
+			w := &cache.ListWatch{
+				WatchFunc: func(options metav1.ListOptions) (watch.Interface, error) {
+					return f.ClientSet.CoreV1().Pods(f.Namespace.Name).Watch(ctx, options)
+				},
+			}
+
+			// Setup watch to continuously monitor any pod events and detect invalid pod status updates
+			_, err = watchtools.Until(ctx, list.ResourceVersion, w, func(event watch.Event) (bool, error) {
+				if pod, ok := event.Object.(*v1.Pod); ok {
+					if isPodStatusAffectedByIssue108594(pod) {
+						return false, fmt.Errorf("failing test due to detecting invalid pod status")
+					}
+					// Watch will never terminate (only when the test ends due to context cancellation)
+					return false, nil
+				}
+				return false, nil
+			})
+
+			// Ignore timeout error since the context will be explicitly cancelled and the watch will never return true
+			if err != nil && !wait.Interrupted(err) {
+				framework.Failf("watch for invalid pod status failed: %v", err.Error())
+			}
+		}()
+
+		ginkgo.By("Verifying batch pods are running")
+		for _, pod := range list.Items {
+			if podReady, err := testutils.PodRunningReady(&pod); err != nil || !podReady {
+				framework.Failf("Failed to start batch pod: %v", pod.Name)
+			}
+		}
+
+		for _, pod := range list.Items {
+			framework.Logf("Pod (%v/%v) status conditions: %q", pod.Namespace, pod.Name, &pod.Status.Conditions)
+		}
+
+		// use to keep the node active before testing critical pods reaching the terminate state
+		delyapodName := "delay-shutdown-20-" + string(uuid.NewUUID())
+		delayPod := getGracePeriodOverrideTestPod(delyapodName, nodeName, 20, scheduling.SystemNodeCritical)
+		e2epod.NewPodClient(f).CreateSync(ctx, delayPod)
+
+		ginkgo.By("Emitting shutdown signal")
+
+		emitSignalPrepareForShutdown(nodeName, f, ctx)
```

Comment:
Is this signal recoverable? Can you reset it somehow on Windows after it was set? Or effectively you can only run a single test per execution?

If it can be reset - I suggest to have `AfterEach` that will reset the signal and restart kubelet to clean up state after each test